src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/FluidBindingProviderExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host
+{    
+    // Internal Extension methods for setting backwards compatibility hooks on certain bindings. 
```

Comment:
I agree with Mathew's feedback from yesterday. It'd be easier to follow if we had some internal IBuildParameterDescriptor/IPostResolveHook interfaces that you could check for, rather than attaching state like this.